test/martha/martha_v3.test.js

```diff
@@ -113,25 +136,20 @@ test.serial('martha_v3 uses the default error handler for unexpected errors', as
 
 // Test the "default" case because we don't know who you are.
 test.serial('martha_v3 resolves a valid DOS-style url', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(sampleDosResponse);
-    getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
+    const bond = bondUrls('dcf-fence');
+    const dos = dosUrls('abc', '123');
+
+    getJsonFromApiStub.withArgs(bond.serviceAccountKeyUrl, terraAuth).resolves(googleSAKeyObject);
+    getJsonFromApiStub.withArgs(dos.dataobjectsUrl, null).resolves(sampleDosResponse);
+
     const response = mockResponse();
     await marthaV3(mockRequest({ body: { 'url': 'dos://abc/123' } }), response);
+
     t.is(response.statusCode, 200);
     const result = response.send.lastCall.args[0];
-    sinon.assert.callCount(getJsonFromApiStub, 2); // Bond was called to get SA key
-    t.deepEqual({ ...result }, sampleDosMarthaResult(googleSAKeyObject));
-    t.is(
-        getJsonFromApiStub.getCall(0).args[0],
-        'https://abc/ga4gh/dos/v1/dataobjects/123',
-    );
-    t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
-    const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(1).args[0];
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'dcf-fence';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
+    t.deepEqual(result, sampleDosMarthaResult(googleSAKeyObject));
+
+    sinon.assert.callCount(getJsonFromApiStub, 2);
```

Comment:
I would rather use sinon mocks (instead of stubs) and `verify()` them here, but I couldn't get mocks to work. Checking the call count, in combination with the defined stub expectations, gets most of what we want. I think it would just be easier to understand failures if we could use `verify()`.